### ISSUE TITLE
Allow screen readers to read all column headers

### ISSUE THREAD

**Issue by GoelBiju** [ral-facilities/datagateway#823]
**Description**:

This issue comes from the [comment](https://github.com/ral-facilities/datagateway/issues/772#issuecomment-911835555) on #813 regarding the accessibility of `datagateway-dataview` table view column headers which do not have sorting/filtering. Since there is no sorting/filtering, a user is unable to tab to the column header and jumps over it completely meaning the screen reader will not state its presence. 

There were various attempts to try this using `tabIndex` in the `DataHeader` component in `datagateway-common` as part of the table view, however, this did not produce correct results with the screen reader with it sometimes introducing more incorrect information.

**Acceptance criteria**:
- [ ] Allow for all column headers to have focus and be read out by the screen reader,

- [ ] Ensure erroneous information is not read out by the screen reader.

**Comment by GoelBiju** [comment]
@MRichards99 had mentioned some issues with the NVDA screen reader on the column headers on a [review comment](https://github.com/ral-facilities/datagateway/pull/813#pullrequestreview-744953210).

The issues will require us to:

- [ ]  ensure that the screen reader only reads out the column header that have focused on and not any other columns before or after it,
- [ ]  ensure when you focus on a new column header it doesn't give the table information (e.g. 'clickable table with 50 rows and 10 columns') every time; this might only be fine only on an initial focus on the table but otherwise you would not want to hear this before every column header is read out.
